src/Symfony/Component/Messenger/Bridge/Amqp/Transport/Connection.php

```diff
@@ -295,7 +295,7 @@ public function publish(string $body, array $headers = [], int $delayInMs = 0, A
         $this->publishOnExchange(
             $this->exchange(),
             $body,
-            $this->getRoutingKeyForMessage($amqpStamp),
+            $forceDefaultRoutingKey ? $this->getDefaultPublishRoutingKey() : $this->getRoutingKeyForMessage($amqpStamp),
```

Comment:
is there a way without adding the new argument, using some state in AmqpStamp?

Reply:
Thanks for your feedback. I followed your recommendations and used AmqpStamp to transport the information.